actor/src/main/scala/org/apache/pekko/io/dns/internal/AsyncDnsResolver.scala

```diff
@@ -151,11 +153,19 @@ private[io] final class AsyncDnsResolver(
     }
 
   private def sendQuestion(resolver: ActorRef, message: DnsQuestion): Future[Answer] = {
-    val result = (resolver ? message).mapTo[Answer]
-    result.failed.foreach { _ =>
-      resolver ! DropRequest(message.id)
+    (resolver ? message).transformWith {
+      case Success(result: Answer) =>
+        Future.successful(result)
+      case Success(DuplicateId(_)) =>
+        sendQuestion(resolver, message.withId(idGenerator.nextId()))
```

Comment:
With random transaction ids its possible the id is already in use. If that is the case retry with a different transaction id